devel/reports.py

```diff
@@ -167,6 +167,20 @@ def non_existing_dependencies(packages):
     return packages
 
 
+def security_packages_overview(packages):
+    filtered = []
+    packages_ids = packages.values_list('id',
+                                       flat=True).order_by().distinct()
+    packages = PackageSecurity.objects.filter(id__in=set(packages_ids))
+    for package in packages:
+        package.pkg.pie = 'PIE enabled' if package.is_pie else 'No PIE'
+        package.pkg.relro = 'Full RELRO' if package.is_relro else 'None'
+        package.pkg.canary = 'Canary found' if package.is_canary else 'No canary found'
+        package.pkg.fortify = 'Yes' if package.is_canary else 'No'
+        filtered.append(package.pkg)
```

Comment:
What about this?

```
    packages = packages.select_related('packagesecurity').order_by()
    for package in packages:
        pkgsec = package.packagesecurity
        package.pie = 'PIE enabled' if pkgsec.is_pie else 'No PIE'
        ...
        filtered.append(package)
```

Reply:
just a reminder that shared object files will not be PIE in the same attribute entry as binary applications are. however DSO objects are still fine when they are shared objects, as they are position independent when dynamically linked in by providing a random base per shared object.

Reply:
This is handled in the class Elf code.
```
          if "ET_DYN" in header['e_type']:
              if self.dynamic_tags("DT_DEBUG"):
                  return True
              return True # DSO is PIE
          return False```